langchain4j-coherence/src/test/java/dev/langchain4j/store/memory/chat/coherence/CoherenceChatMemoryStoreIT.java

```diff
@@ -0,0 +1,144 @@
+import com.oracle.bedrock.runtime.options.DisplayName;
+import com.oracle.bedrock.testsupport.junit.TestLogsExtension;
+import com.tangosol.net.Session;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.Content;
+import dev.langchain4j.data.message.ImageContent;
+import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.UserMessage;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class CoherenceChatMemoryStoreIT {
+
+    @RegisterExtension
+    static TestLogsExtension testLogs = new TestLogsExtension();
+
+    @RegisterExtension
+    static CoherenceClusterExtension cluster = new CoherenceClusterExtension()
+            .with(ClusterName.of("NamedMapEmbeddingRepositoryIT"),
+                    WellKnownAddress.loopback(),
+                    LocalHost.only(),
+                    IPv4Preferred.autoDetect(),
+                    SystemProperty.of("coherence.serializer", "pof"))
+            .include(3, CoherenceClusterMember.class,
+                    DisplayName.of("storage"),
+                    RoleName.of("storage"),
+                    testLogs);
+
+    static Session session;
+
+    private final String userId = "someUserId";
+
+    private CoherenceChatMemoryStore memoryStore;
+
+    @BeforeAll
+    public static void beforeAll() {
+        session = cluster.buildSession(SessionBuilders.storageDisabledMember(RoleName.of("test")));
+    }
+
+    @BeforeEach
+    void setUp(TestInfo testInfo) {
+        this.memoryStore = CoherenceChatMemoryStore.builder()
+                .session(session)
+                .name(testInfo.getDisplayName())
+                .build();
+        memoryStore.deleteMessages(userId);
+        List<ChatMessage> messages = memoryStore.getMessages(userId);
+        assertThat(messages).isEmpty();
+    }
+
+    @Test
+    void should_set_messages_into_coherence() {
+        // given
+        List<ChatMessage> messages = memoryStore.getMessages(userId);
+        assertThat(messages).isEmpty();
+
+        // when
+        List<ChatMessage> chatMessages = new ArrayList<>();
+        chatMessages.add(new SystemMessage("You are a large language model working with Langchain4j"));
+        List<Content> userMsgContents = new ArrayList<>();
+        userMsgContents.add(new ImageContent("someCatImageUrl"));
+        chatMessages.add(new UserMessage("What do you see in this image?", userMsgContents));
+        memoryStore.updateMessages(userId, chatMessages);
+
+        // then
+        messages = memoryStore.getMessages(userId);
+        assertThat(messages).hasSize(2);
```

Comment:
Could you please add a bit more assertions? E.g. that returned messages are identical to persisted ones